File: backend/src/main/java/com/chatapp/dto/AuthResponse.java
package com.chatapp.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Response containing authentication token and user information")
public class AuthResponse {

    @Schema(
            description = "JWT authentication token to be used in Authorization header",
            example = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
    )
    private String token;

    @Schema(
            description = "Type of token (always 'Bearer')",
            example = "Bearer",
            defaultValue = "Bearer"
    )
    private String type = "Bearer";

    @Schema(
            description = "Unique identifier of the authenticated user",
            example = "1"
    )
    private Long userId;

    @Schema(
            description = "Username of the authenticated user",
            example = "john_doe"
    )
    private String username;

    @Schema(
            description = "Email of the authenticated user",
            example = "john.doe@example.com"
    )
    private String email;

    @Schema(
            description = "Display name of the authenticated user",
            example = "John Doe"
    )
    private String displayName;
}

File: client/composeApp/src/androidMain/kotlin/com/chatapp/client/data/model/AuthResponse.kt
package com.chatapp.client.data.model

import kotlinx.serialization.Serializable

@Serializable
data class AuthResponse(
    val token: String,
    val type: String = "Bearer",
    val userId: Long,
    val username: String,
    val email: String,
    val displayName: String?
)

@Serializable
data class ApiResponse<T>(
    val success: Boolean,
    val data: T? = null,
    val message: String? = null,
    val error: String? = null,
    val timestamp: String? = null
)